management/management_request_test.go

```diff
@@ -0,0 +1,91 @@
+package management
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewRequest(t *testing.T) {
+	testCases := []struct {
+		name          string
+		method        string
+		uri           string
+		payload       interface{}
+		options       []RequestOption
+		expectedError error
+	}{
+		{
+			name:          "Valid GET request",
+			method:        "GET",
```

Comment:
Any chance we could use the constants from the `http` package for the methods? `http.MethodGet`